src/main/java/prefs/Preferences.java

```diff
@@ -18,13 +17,8 @@
 
     public GlobalConfig global;
 
-    public Preferences(boolean isTestMode) {
-        if (isTestMode) {
-            this.fileHandler = new ConfigFileHandler(DIRECTORY, TEST_CONFIG_FILE);
-        } else {
-            this.fileHandler = new ConfigFileHandler(DIRECTORY, GLOBAL_CONFIG_FILE);
-        }
-
+    public Preferences(Optional<String> configFileName) {
```

Comment:
Good point. I think in that case we should just ask for a `String` without an `Optional`.